### PR TITLE
WIP: Feature xml activities

### PR DESCRIPTION
**Pull request by Andale-online**
Better PR for checking #152

### CODE REVIEW

File: src/activities/Elsa.Activities.Http/Activities/WriteXmlHttpResponse.cs [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+
+        protected override async Task<ActivityExecutionResult> OnExecuteAsync(
+            WorkflowExecutionContext workflowContext,
+            CancellationToken cancellationToken)
+        {
+            var response = httpContextAccessor.HttpContext.Response;
+
+            if (response.HasStarted)
+                return Fault("Response has already started");
```
Comment (Andale-online) [inline src/activities/Elsa.Activities.Http/Activities/WriteXmlHttpResponse.cs]: This fails, because response.HasStarted == true

**Comment by sfmskywalker** [comment]
I'm not at all convinced we need an almost-duplicate version of the general-purpose `ReceiveHttpRequest` just to handle XML.

The need to be able to match a root XML element name against a .NET type seems too application-specific.

IMO it would be better if the `ReceiveHttpRequest` itself would be able to receive the XML payload, just like it can receive JSON payloads and Form payloads.

Other activities can then access these payloads in their native forms:

* JSON as JObject object (converted to ExpandoObject to allow dot notation)
* XML as XElement
* Form as key/value pairs (dictionary of StringValues)

I do see one complimentary activity type that will meet your need:

* BindModel

`BindModel` would have one `ModelType:Type` property` and one `Expression:WorkflowExpression` property evaluating to the value to be bound against.

It would evaluate the expression and inspect the value type returned. If its JSON, it will deserialize it into the specified type. If its XML, same thing. And of course same story for the Form values. In fact, I think we might be able to reuse actual MVC model binding for this.

Regarding the `AssemblyName` property and the need to actually load the DLL: why is this necessary? Aren't all the types you use already part of the app domain if you reference these assemblies from your application project (directly or indirectly)?

Alternatively, you could consider changing the `ReceiveXmlHttpRequest` into a `BindHttpRequest` activity that does the model binding I described.

Let me know if you would like to discuss this further over Skype or anything. Alternatively, I'm happy to implement what I proposed myself. Meanwhile you can of course use the proposed activity as-is.

**Comment by Andale-online** [comment]
I think we'll discuss this later. Do you have an idea about the response already started thingy?

**Comment by Andale-online** [comment]
Stop thinking about it... I think i got it...

**Comment by Andale-online** [comment]
We can close this... I'll submit a "bug report"

**Pull request closed without merging**